Code/GraphMol/Wrap/Mol.cpp

```diff
@@ -89,9 +89,11 @@ namespace RDKit {
   }
   PyObject *GetSubstructMatch(const ROMol &mol, const ROMol &query,bool useChirality=false,
                             bool useQueryQueryMatches=false){
-    NOGIL gil;
     MatchVectType matches;
-    SubstructMatch(mol,query,matches,true,useChirality,useQueryQueryMatches);
+    {
+      NOGIL gil;
```

Comment:
How about modifying the NOGIL definition so that it only does anything if the thread safety flag (RDK_BUILD_THREADSAFE_SSS) is set?